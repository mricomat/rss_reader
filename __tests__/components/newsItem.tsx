import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';

import MovieItem from 'src/components/newsItem';

const onPress = jest.fn();

describe('test Input', () => {
  test('it renders correctly', () => {
    const input = render(<MovieItem />);
    expect(input).toMatchSnapshot();
  });

  test('it renders correctly with title and description', () => {
    const input = render(<MovieItem title={'test'} description={'descriptionTest'} />);
    expect(input).toMatchSnapshot();
  });

  test('onPress is called correctly', () => {
    const { getByTestId } = render(<MovieItem title={'test'} description={'descriptionTest'} onPress={onPress} />);
    fireEvent.press(getByTestId('newsItem'));
    expect(onPress).toHaveBeenCalled();
  });
});
