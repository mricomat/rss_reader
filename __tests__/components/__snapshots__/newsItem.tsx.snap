// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Input it renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderRadius": 8,
      "flex": 1,
      "minHeight": 140,
      "opacity": 1,
    }
  }
  testID="newsItem"
>
  <Image
    source={
      Object {
        "uri": undefined,
      }
    }
    style={
      Object {
        "borderRadius": 8,
        "height": "100%",
        "opacity": 0.2,
        "position": "absolute",
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "flexDirection": "row",
        "height": "100%",
        "padding": 10,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "borderRadius": 6,
            "height": "100%",
            "width": 80,
          },
          Object {
            "backgroundColor": "#DADADA",
            "margin": 10,
            "position": "absolute",
          },
        ]
      }
    />
    <Image
      source={
        Object {
          "uri": undefined,
        }
      }
      style={
        Object {
          "borderRadius": 6,
          "height": "100%",
          "width": 80,
        }
      }
    />
    <View
      style={
        Object {
          "flex": 1,
          "paddingHorizontal": 6,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontSize": 18,
            "fontWeight": "600",
          }
        }
      />
      <Text
        numberOfLines={2}
        style={
          Object {
            "color": "#FFFFFF",
            "fontSize": 16,
            "fontWeight": "400",
            "marginTop": 8,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`test Input it renders correctly with title and description 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderRadius": 8,
      "flex": 1,
      "minHeight": 140,
      "opacity": 1,
    }
  }
  testID="newsItem"
>
  <Image
    source={
      Object {
        "uri": undefined,
      }
    }
    style={
      Object {
        "borderRadius": 8,
        "height": "100%",
        "opacity": 0.2,
        "position": "absolute",
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "flexDirection": "row",
        "height": "100%",
        "padding": 10,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "borderRadius": 6,
            "height": "100%",
            "width": 80,
          },
          Object {
            "backgroundColor": "#DADADA",
            "margin": 10,
            "position": "absolute",
          },
        ]
      }
    />
    <Image
      source={
        Object {
          "uri": undefined,
        }
      }
      style={
        Object {
          "borderRadius": 6,
          "height": "100%",
          "width": 80,
        }
      }
    />
    <View
      style={
        Object {
          "flex": 1,
          "paddingHorizontal": 6,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontSize": 18,
            "fontWeight": "600",
          }
        }
      >
        test
      </Text>
      <Text
        numberOfLines={2}
        style={
          Object {
            "color": "#FFFFFF",
            "fontSize": 16,
            "fontWeight": "400",
            "marginTop": 8,
          }
        }
      >
        descriptionTest
      </Text>
    </View>
  </View>
</View>
`;
